### PR TITLE
[vcpkg] Add `vcpkg export` to E2E tests. Enable E2E tests on all platforms.

### PR DESCRIPTION
**Pull request by ras0219**
See title.

In order to enable compartmentalized testing, also adds `vcpkg export --output-dir=/export/artifacts/`. This is implemented for all first-party export mechanisms, however I have declined to implement it for experimental community export mechanisms because we do not have reliable testing for them.

### CODE REVIEW

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Don't forget to remove new vcpkg_downloads references when resolving the merge

**Comment by BillyONeal** [comment]
@ras0219 @ras0219-msft @strega-nil e2e99d0 contains 'interesting' merge conflict resolutions from #12314

File: toolsrc/src/vcpkg/base/files.cpp [hunk 1]
```diff
@@ -930,7 +930,7 @@ namespace vcpkg::Files
                 return res;
             }
 
-            System::printf("Waiting to take filesystem lock on %s...\n", path.u8string());
+            Debug::print("Waiting to take filesystem lock on ", path.u8string(), "...\n");
```
Comment (strega-nil) [inline toolsrc/src/vcpkg/base/files.cpp:933]: I don't know if this is correct; what's the argument for changing this?

**Review by strega-nil — APPROVED** [review]
I'm curious about the one thing; also, do `variables` automagically get turned into environment variables?

File: toolsrc/src/vcpkg/base/files.cpp (hunk 1 above)
Comment (BillyONeal) [reply toolsrc/src/vcpkg/base/files.cpp:933]: It's a verbosity reduction to be debug only rather than always.

**Comment by BillyONeal** [comment]
> I'm curious about the one thing; also, do `variables` automagically get turned into environment variables?

Yes: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch

>System and user-defined variables also get injected as environment variables for your platform. When variables are turned into environment variables, variable names become uppercase, and periods turn into underscores. For example, the variable name `any.variable` becomes the variable name `$ANY_VARIABLE`.

File: toolsrc/src/vcpkg/base/files.cpp (hunk 1 above)
Comment (strega-nil) [reply toolsrc/src/vcpkg/base/files.cpp:933]: hmm, I guess. It should be a printf for `--x-wait-for-lock` though, I think.

Comment (ras0219-msft) [reply toolsrc/src/vcpkg/base/files.cpp:985]: This caused excess output which prevented:
```
& $(./vcpkg fetch cmake) --build .
```
in powershell

**Review by strega-nil — APPROVED** [review]

**Pull request merged**